packages/dds/matrix/src/test/time/sharedMatrix.spec.ts

```diff
@@ -0,0 +1,282 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import assert from "node:assert";
+
+import {
+	benchmark,
+	BenchmarkType,
+	isInPerformanceTestingMode,
+	type BenchmarkTimer,
+} from "@fluid-tools/benchmark";
+
+import { SharedMatrix } from "../../index.js";
+import { createLocalMatrix } from "../utils.js";
+
+describe("SharedMatrix execution time", () => {
+	// The value to be set in the cells of the matrix.
+	const matrixValue = "cellValue";
+	// The test matrix's size will be 10*10, 100*100, 1000*1000.
+	const matrixSizes = isInPerformanceTestingMode
+		? [10, 100, 1000]
+		: // When not measuring perf, use a single smaller data size so the tests run faster.
+			[10];
+
+	// The number of operations to perform on the matrix.
+	const operationCounts = isInPerformanceTestingMode
+		? [10, 100, 1000]
+		: // When not measuring perf, use a single smaller data size so the tests run faster.
+			[5];
+
+	let localMatrix: SharedMatrix;
+	for (const matrixSize of matrixSizes) {
```

Comment:
All of the actual logic being measured here is the same as in the memory tests, right? Can we extract the test cases into a library of functions so we can shared code between the two test scenarios (memory and runtime)?

Reply:
I'm not sure if it'd be worth it... in `benchmarkFnCustom` we have to do some things "between" the pieces of what could be shared test logic (e.g. between creating the matrix and then running operations on it, so we can't easily wrap all the shared test logic neatly). My gut feeling is that the way in ends up looking if we do it, wouldn't be very readable. In a hypothetical future where the infrastructure for memory and runtime benchmarks is more similar, I could see it.